obal/data/roles/setup-sources/tasks/main.yml

```diff
@@ -0,0 +1,59 @@
+---
+- set_fact:
+    package_dir: "{{ inventory_dir }}/packages/{{ inventory_hostname }}"
+    setup_sources_annex_options: setup_sources_annex_options
+  tags:
+    - always
+
+- name: get spec file
+  find:
+    pattern: "*.spec"
+    path: "{{ package_dir }}"
+  register: spec_file
+  tags:
+    - always
+
+- set_fact:
+    spec_file_path: "{{ spec_file.files[0].path }}"
+  tags:
+    - always
+
+- name: 'Extract Source0 from spec'
+  shell: "spectool --list-files --source 0 {{ spec_file_path }} | awk '{print $2}'"
+  register: source0
+  changed_when: False
+  tags:
+    - always
+
+- set_fact:
+    source0: "{{ source0.stdout }}"
+    sourcebase: "{{ source0.stdout | basename }}"
+  tags:
+    - always
+
+- set_fact:
+    sourcepath: "{{ package_dir  }}/{{ sourcebase }}"
+  tags:
+    - always
+
+- name: 'Source exists'
+  stat: 
+    path: "{{ sourcepath }}"
+  register: source_exists
+
+- name: 'Search annex for web link'
+  shell: 'git annex whereis {{ sourcepath }} 2>/dev/null | grep -q " web"'
+  args:
+    chdir: "{{ package_dir }}"
+  changed_when: False
+  register: search_annex
+  ignore_errors: yes
+  when: setup_sources_force or ( source_exists.stat.islink is defined and source_exists.stat.islnk )
+
+- name: 'Add url to annex for source'
+  shell: 'git annex addurl {{ setup_sources_annex_options }} --file "{{ sourcebase }}" "{{ source0 }}"'
+  args:
+    chdir: "{{ package_dir }}"
+  ignore_errors: yes
+  when: search_annex.rc is defined and search_annex.rc != 0
```

Comment:
I'm still quite new to ansible but would ```search_annex|failed``` work?

Reply:
Yup.